### DISCUSSION TITLE
Instructor accounts vs superadmins?

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#951] [Q&A] [answered]
Are super admins the only users who can create new courses? I wondered if there was a teacher account setting where they could create/admin/delete courses they're responsible for but not see all the courses. No mention of it in the documentation seems like users can be added to a group for a course if created as teachers, but no like the 'teacher admin' role differentiation from a superadmin. 

Just wanted to confirm, or if such roles exist, understand how they are created. Thanks.

**Comment by anthonygego** [comment, marked as answer]
Indeed, for several historical reasons. Currently only the superadmins can create courses. 

The few roles currently existing in the app (superadmin, course admin, tutor and student) are implemented more as flags than part of a role mechanism and would require a small refactor to evolve to something more robust. But part of the job is already done at the UI side as a global admin panel is being introduced (historically we only supported external identity providers), which would allow to specify roles more easily.

This should come in the next few months (I hope...) as we are investigating solutions to make our courses material re-usable for high school teachers, with limited IT knowledge, in their own classroom context, in a transparent way, and a more fine-grained role/permission mechanism is clearly part of that job.
